querybook/webapp/ui/Comment/Comments.tsx

```diff
@@ -159,50 +195,68 @@ export const Comments: React.FunctionComponent<IProps> = ({
-                        setCurrentComment(editorState.getCurrentContent())
-                    }
-                    placeholder={commentIds.length ? 'Reply' : 'Comment'}
-                />
-                <IconButton
-                    icon="XCircle"
-                    onClick={handleCommentClear}
-                    noPadding
-                    size={18}
-                    className="mr12"
-                    tooltip="Clear"
-                    tooltipPos="left"
-                    disabled={
-                        currentComment.getPlainText().length === 0 &&
-                        editingCommentId === null
-                    }
-                />
-                <IconButton
-                    icon="Send"
-                    onClick={handleCommentSave}
-                    noPadding
-                    size={18}
-                    className="mr4"
-                    tooltip="Comment"
-                    tooltipPos="left"
-                    disabled={currentComment.getPlainText().length === 0}
-                />
-            </div>
+            {commentIds ? (
+                <>
+                    <div className="Comments-list p16">
+                        {commentIds.length ? (
+                            renderCommentDOM()
+                        ) : (
+                            <EmptyText>No Comments</EmptyText>
+                        )}
+                    </div>
+                    {editingCommentId ? renderEditingCommentWarning() : null}
+                    <div className="Comment-form flex-row pv12 ph16">
+                        <Formik
+                            enableReinitialize
+                            initialValues={{ text: currentComment }}
+                            onSubmit={handleCommentSave}
+                        >
+                            {({ submitForm, values, setValues }) => (
+                                <>
+                                    <UserAvatar uid={userInfo?.uid} tiny />
+                                    <RichTextField name="text" />
+                                    <IconButton
+                                        icon="XCircle"
+                                        onClick={() =>
+                                            handleCommentClear(() =>
+                                                setValues({
+                                                    text: emptyCommentValue,
+                                                })
+                                            )
+                                        }
+                                        noPadding
+                                        size={18}
+                                        className="mr12"
+                                        tooltip="Clear"
+                                        tooltipPos="left"
+                                        disabled={
+                                            values.text.getPlainText()
```

Comment:
can you set this as variable instead of checking it twice? not sure if this can be costly operation